### PR TITLE
Check before overwriting server config

### PR DESCRIPTION
**Pull request by calvix**
Override server factory config only if it has empty value

### CODE REVIEW

File: command/daemon/command.go [hunk 1]
```diff
@@ -122,10 +122,18 @@ func (c *command) Execute(cmd *cobra.Command, args []string) {
...
+		}
+		if serverConfig.TLSKeyFile == "" {
+			serverConfig.TLSKeyFile = c.viper.GetString(f.Server.TLS.KeyFile)
+		}
+		if serverConfig.TLSCAFile == "" {
+			serverConfig.TLSCAFile = c.viper.GetString(f.Server.TLS.CaFile)
+		}
+		if serverConfig.ListenAddress == "" {
+			serverConfig.ListenAddress = c.viper.GetString(f.Server.Listen.Address)
+		}
```
Comment (xh3b4sd) [inline command/daemon/command.go]: Nitpick, I would like to keep the order of the properties as it was before. Order was alphabetically.

**Comment by calvix** [comment]
@xh3b4sd fixed

**Comment by calvix** [comment]
test are not passing becasue it expect non-empty listenAddress in config when running `New`

File: server/server.go [hunk 2]
```diff
@@ -124,9 +124,6 @@ func New(config Config) (Server, error) {
 	if config.HandlerWrapper == nil {
 		return nil, microerror.Maskf(invalidConfigError, "handler wrapper must not be empty")
 	}
-	if config.ListenAddress == "" {
-		return nil, microerror.Maskf(invalidConfigError, "listen address must not be empty")
-	}
```
Comment (xh3b4sd) [inline server/server.go]: This check must not be removed. When tests fail they should rather be adapted to have a listen address being configured.

**Comment by calvix** [comment]
Thanks for guide, fixed.

**Pull request merged**